src/internal/m365/collection/exchange/collection.go

```diff
@@ -252,17 +252,25 @@ func (col *prefetchCollection) streamItems(
 	errs *fault.Bus,
 ) {
 	var (
+		category    path.CategoryType
 		success     int64
 		totalBytes  int64
 		wg          sync.WaitGroup
 		colProgress chan<- struct{}
 
 		user = col.user
-		log  = logger.Ctx(ctx).With(
-			"service", path.ExchangeService.String(),
-			"category", col.FullPath().Category().String())
 	)
 
+	if col.FullPath() != nil {
+		category = col.FullPath().Category()
```

Comment:
no problems with this implementation, but do you think it'd be worth moving this to the base collection as a `col.Category()` func?  Right now it's up to every caller to check this same pattern.

Reply:
that's an excellent idea. Since the issue is already in main though, let me queue this to avoid long-term issues (in case we forget about anything) then we can get a nicer solution added to BaseCollection